staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -390,6 +390,7 @@ func (r *Request) Body(obj interface{}) *Request {
 	case []byte:
 		glogBody("Request Body", t)
 		r.body = bytes.NewReader(t)
+		r.SetHeader("Content-Type", "application/json")
```

Comment:
There are many different kind of bytes (json, yaml, proto, opaque).  Rather than set this here, you can set it about here https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/dynamic/simple.go#L113 with a `SetHeader` call.

Once you get this updated (or a new PR created), direct message me on slack.